tests/test_solvetransposed3x3.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+  Copyright 2021 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+
+#define BOOST_TEST_MODULE SolveTransposed3x3
+#include <boost/test/unit_test.hpp>
+#include <boost/test/floating_point_comparison.hpp>
+
+#include <dune/istl/bcrsmatrix.hh>
+
+#include <opm/simulators/linalg/bda/CPR.hpp>
+
+BOOST_AUTO_TEST_CASE(testsolvetransposed3x3)
+{
+    const unsigned numTests = 3;
+    const unsigned blockSize = 3;
+
+    std::vector<std::vector<double> > A = {{-0.0973322, 1.18758e-08, -0.000185231,
+                                            0.125114, 1.24919e-11, 0,
+                                            -11.3854, 1.32995e-06, 0.065447},
+                                           {-0.00354818, 2.24354e-08, 55.9563,
+                                            0.284031, 2.83543e-11, 0,
+                                            -62.1156, 0.000392767, 6350.26},
```

Comment:
I am a bit unsure whether we really should use these test matrices. They are really ill-conditioned and we cannot assume that our computed result is correct. Yet we are assuming our algorithm (and test data) to be exact as we allow only 10^-6% difference. 

If I take e.g. the second matrix and use octave to solve it, the second entry of the solution has opposite sign.
```
C=[ -0.00354818, 2.24354e-08, 55.9563; 0.284031, 2.83543e-11, 0; 62.1156, 0.000392767, 6350.26];
linsolve(C', [0;1;0])
```
```
ans =

  -2.9083e+05
  -5.6408e+05
   2.5627e+03
```


maybe we should use artificially generated matrices (a bit less ill conditioned) and another program for computing the reference solution?

Reply:
Do you mean generate randoms with Python or something like that? I'm not sure where to get them.

Reply:
I meant you as a random number generator should think of a few 3x3 ones. They should not be to random, not symmetric, invertible, use values that are equal to integers or fractions.

Reply:
I changed the testcases, and also solve them via Dune.